url9582d546a9_twentylemon_sudoku_generator_tgz-pJ8-sudoku_Top95J8/src/sudoku/util/ArrayUtil.java

```diff
@@ -0,0 +1,62 @@
+package sudoku.util;
+
+import java.util.Random;
+
+/**
+ * Provides utility functions for arrays.
+ *
+ * @author Taras Mychaskiw
+ */
+public class ArrayUtil {
+
+    private final static Random rand = new Random();
+
+    private ArrayUtil(){}
+
+    /**
+     * Returns an array of all integers [start..end) exclusive.
+     *
+     * @param start the value at the first position in the array
+     * @param end the value at the last position in the array
+     * @return the array of integers [start..end)
+     */
+    public static int[] range(int start, int end){
+        final int size = end - start;
+        final int[] array = new int[size];
+        for (int i = 0; i < array.length; i++){
+            array[i] = i + start;
+        }
+        return array;
+    }
+
+    /**
+     * Shuffles the array given.
+     *
+     * @param array the array to shuffle
+     */
+    public static void shuffle(int[] array){
+        for (int i = 0; i < array.length - 1; i++){
+            int j = rand.nextInt(i + 1);
+            int temp = array[j];
+            array[j] = array[i];
+            array[i] = temp;
+        }
+    }
+
+    /**
+     * Returns true if the value is in the array given.
+     *
+     * @param <T> the type held in the array
+     * @param array the array to search in
+     * @param value the value to search for
+     * @return true if the value is in the array
+     */
+    public static <T> boolean contains(T[] array, T value){
+        for (T entry : array){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;array&#39; (index: 0 | type: T[). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;value&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).